When running HAPI-FHIR in any binary mode outside of DATABASE, package loading fails.

```
Caused by: java.lang.RuntimeException: Failed to load package resource NpmPackageVersionResourceEntity[myId=1,myCanonicalUrl=http://hl7.org/fhir/us/core/StructureDefinition/head-occipital-frontal-circumference-percentile,myCanonicalVersion=3.1.1,myResourceType=StructureDefinition,myDirectory=package,myFilename=StructureDefinition-head-occipital-frontal-circumference-percentile.json,myPackageVersion=NpmPackageVersionEntity[myId=1,myPackageId=hl7.fhir.us.core,myVersionId=3.1.1,myDescriptionUpper=<null>,myFhirVersionId=4.0.1],myResSizeBytes=128351,myVersion=2021-10-08 18:07:11.375]
	at ca.uhn.fhir.jpa.packages.JpaPackageCache.loadPackageEntity(JpaPackageCache.java:525)
	at ca.uhn.fhir.jpa.packages.JpaPackageCache.lambda$loadPackageAssetsByType$10(JpaPackageCache.java:698)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$Itr.forEachRemaining(ArrayList.java:1033)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at ca.uhn.fhir.jpa.packages.JpaPackageCache.loadPackageAssetsByType(JpaPackageCache.java:698)
	at jdk.internal.reflect.GeneratedMethodAccessor394.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at ca.uhn.fhir.jpa.packages.NpmJpaValidationSupport.fetchAllStructureDefinitions(NpmJpaValidationSupport.java:79) [10 skipped]
	at ca.uhn.fhir.context.support.IValidationSupport.fetchAllNonBaseStructureDefinitions(IValidationSupport.java:119)
	at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain.lambda$fetchAllNonBaseStructureDefinitions$1(ValidationSupportChain.java:205)
	at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain.doFetchStructureDefinitions(ValidationSupportChain.java:212)
	at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain.fetchAllNonBaseStructureDefinitions(ValidationSupportChain.java:205)
	at org.hl7.fhir.common.hapi.validation.support.BaseValidationSupportWrapper.fetchAllNonBaseStructureDefinitions(BaseValidationSupportWrapper.java:46)
	at org.hl7.fhir.common.hapi.validation.support.CachingValidationSupport.lambda$fetchAllNonBaseStructureDefinitions$2(CachingValidationSupport.java:90)
	at org.hl7.fhir.common.hapi.validation.support.CachingValidationSupport.lambda$loadFromCache$14(CachingValidationSupport.java:157)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2031)
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1908)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2029)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2012)
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:113)
	at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:54)
	at org.hl7.fhir.common.hapi.validation.support.CachingValidationSupport.loadFromCache(CachingValidationSupport.java:158)
	at org.hl7.fhir.common.hapi.validation.support.CachingValidationSupport.fetchAllNonBaseStructureDefinitions(CachingValidationSupport.java:90)
	at ca.uhn.fhir.rest.server.provider.ServerCapabilityStatementProvider.getSupportedProfileMultimap(ServerCapabilityStatementProvider.java:567)
	at ca.uhn.fhir.rest.server.provider.ServerCapabilityStatementProvider.getServerConformance(ServerCapabilityStatementProvider.java:200)
	at ca.uhn.fhir.rest.server.provider.ServerCapabilityStatementProvider.getServerConformance(ServerCapabilityStatementProvider.java:90)
	at ca.cdr.endpoint.fhir.web.CdrConformanceProvider.getServerConformance(CdrConformanceProvider.java:58)
	at jdk.internal.reflect.GeneratedMethodAccessor398.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at ca.uhn.fhir.rest.server.method.BaseMethodBinding.invokeServerMethod(BaseMethodBinding.java:264) [1 skipped]
	at ca.uhn.fhir.rest.server.method.ConformanceMethodBinding.createCapabilityStatement(ConformanceMethodBinding.java:178)
	at ca.uhn.fhir.rest.server.method.ConformanceMethodBinding.invokeServer(ConformanceMethodBinding.java:171)
	at ca.uhn.fhir.rest.server.method.ConformanceMethodBinding.invokeServer(ConformanceMethodBinding.java:54)
	at ca.uhn.fhir.rest.server.method.BaseResourceReturningMethodBinding.doInvokeServer(BaseResourceReturningMethodBinding.java:326)
	at ca.uhn.fhir.rest.server.method.BaseResourceReturningMethodBinding.invokeServer(BaseResourceReturningMethodBinding.java:434)
	at ca.uhn.fhir.rest.server.RestfulServer.handleRequest(RestfulServer.java:1152)
	at ca.uhn.fhir.rest.server.RestfulServer.doGet(RestfulServer.java:413)
	... 125 common frames omitted
Caused by: java.lang.UnsupportedOperationException: null
	at ca.uhn.fhir.jpa.binstore.NullBinaryStorageSvcImpl.fetchDataBlobFromBinary(NullBinaryStorageSvcImpl.java:89)
	at ca.uhn.fhir.jpa.packages.JpaPackageCache.fetchBlobFromBinary(JpaPackageCache.java:202)
	at ca.uhn.fhir.jpa.packages.JpaPackageCache.loadPackageEntity(JpaPackageCache.java:520)
	... 167 common frames omitted
```

--- a/hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/5_7_0/3198-package-loading-woes.yaml
+++ b/hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/5_7_0/3198-package-loading-woes.yaml
@@ -1,0 +1,5 @@
+---
+type: change
+issue: 3198
+jira: SMILE-3452
+title: "Fixed a regression where packages would fail to load if HAPI-FHIR was operating in DATABASE binary storage mode."

--- a/hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/packages/JpaPackageCache.java
+++ b/hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/packages/JpaPackageCache.java
@@ -28,6 +28,7 @@ import ca.uhn.fhir.jpa.api.dao.DaoRegistry;
 import ca.uhn.fhir.jpa.api.dao.IFhirResourceDao;
 import ca.uhn.fhir.jpa.api.model.ExpungeOptions;
 import ca.uhn.fhir.jpa.binstore.IBinaryStorageSvc;
+import ca.uhn.fhir.jpa.binstore.NullBinaryStorageSvcImpl;
 import ca.uhn.fhir.jpa.dao.data.INpmPackageDao;
 import ca.uhn.fhir.jpa.dao.data.INpmPackageVersionDao;
 import ca.uhn.fhir.jpa.dao.data.INpmPackageVersionResourceDao;
@@ -198,7 +199,7 @@ public class JpaPackageCache extends BasePackageCacheManager implements IHapiPac
 	 * @throws IOException
 	 */
 	private byte[] fetchBlobFromBinary(IBaseBinary theBinary) throws IOException {
-		if (myBinaryStorageSvc != null) {
+		if (myBinaryStorageSvc != null && !(myBinaryStorageSvc instanceof NullBinaryStorageSvcImpl)) {
 			return myBinaryStorageSvc.fetchDataBlobFromBinary(theBinary);
 		} else {
 			byte[] value = BinaryUtil.getOrCreateData(myCtx, theBinary).getValue();

